api/lib/opentelemetry.rb

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module OpenTelemetry
+  extend self
+
+  def self.tracer
+    instance.tracer
+  end
+
+  def self.meter
+    instance.meter
+  end
+
+  def self.distributed_context_manager
+    instance.distributed_context_manager
+  end
+
+  private
+
+  def self.instance
+    # TODO how to configure provider? https://github.com/open-telemetry/opentelemetry-specification/issues/39
```

Comment:
I have no idea how this should be done. What's the "Ruby way" to do this? Suggestions welcome! The idea is for an application to configure a `tracer`, a `meter` and a `distributed_context_manager`, or default to the implementations provided in the API gem (i.e. no-op implementations, other the trace context propagation).

Reply:
In OpenTracing we did https://github.com/opentracing/opentracing-ruby/blob/master/lib/opentracing.rb#L35-L45 . Some projects define `configure()` method

Reply:
OpenCensus has a ` configure ` method like:
```ruby
OpenCensus::Trace.configure do |config|
  config.default_sampler = OpenCensus::Trace::Samplers::RateLimiting.new
  config.default_max_attributes = 16
end
```
https://github.com/census-instrumentation/opencensus-ruby/blob/master/lib/opencensus/trace/config.rb
The concern there is the additional indirection to lookup the tracer or, if caching, the risk of use-before-configure leading to caching the default tracer. Not sure either are legitimate concerns. 

The OpenTracing solution is much simpler, but permits the tracer to be replaced at any time.